llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/MaterializeEncoding.cpp

```diff
@@ -66,7 +66,8 @@ getMaterializedType(RankedTensorType tensorType,
     return dropEncoding(tensorType);
   }
   return tensor::PackOp::inferPackedType(
-             getOriginalTypeWithEncoding(tensorType),
+             getOriginalTypeWithEncoding(tensorType)
+                 .clone(tensorType.getElementType()),
```

Comment:
I am not sure what this line is doing?

Reply:
This takes the shape from the `original_type` attribute on the encoding, and the element type from `tensorType`. We shouldn't use the element type of `original_type` because it is the type of the unpacked input, and the result type of any CastOpInterface will have a different element type.